src/tests/test_config.py

```diff
@@ -157,12 +157,12 @@ def test_config_deserialization_raises_error(self):
             CompilerConfig.create_from_json(serialized_data)
 
     def test_json_version_compatibility(self):
-        serialised_data = _get_contents("serialised_default_compiler_config_v1.json")
+        serialised_data = _get_contents("serialised_default_compiler_config_v2.json")
         deserialised_conf = CompilerConfig.create_from_json(serialised_data)
         assert deserialised_conf.metrics == MetricsType.Default
         assert deserialised_conf.results_format == QuantumResultsFormat()
 
-        serialised_data = _get_contents("serialised_full_compiler_config_v1.json")
+        serialised_data = _get_contents("serialised_full_compiler_config_v2.json")
```

Comment:
You actually need to add new tests, because you still want to know if v1 serializes correctly.

Reply:
Well if that's the case then you're going to need to do some work to properly map them from v1 to v2 as a part of this PR and make sure you can transmute v1 to v2, since nothing you've done here is un-transmutable. 

I can add this to the PR if you don't have time, but this can't be put in with the capability to transmute from v1 to v2.

Reply:
Yes this makes sense - I have very low capacity at the moment but can try to get to this some point in middle of next week if you do not have time to add this before :)